pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarRecordCursor.java

```diff
@@ -347,8 +355,16 @@ public void run() {
 
                             entriesProcessed += entriesToSkip;
                         } else {
-                            outstandingReadsRequests.decrementAndGet();
-                            cursor.asyncReadEntries(batchSize, this, System.nanoTime(), PositionImpl.latest);
+                            long maxSizeBytes = entryCacheSizeAllocator.getAvailableCacheSize();
+                            // if the available size is invalid and the entry queue size is 0, read one entry
+                            if (maxSizeBytes > 0 || entryQueue.size() == 0) {
+                                outstandingReadsRequests.decrementAndGet();
+                                cursor.asyncReadEntries(batchSize, maxSizeBytes,
+                                        this, System.nanoTime(), PositionImpl.latest);
+                            } else {
+                                metricsTracker.incr_READ_ATTEMPTS_FAIL();
```

Comment:
If people doesn't configure this setting, we are initializing a NullCacheSizeAllocator. So it will cause `metricsTracker` to increase `READ_ATTEMPTS_FAIL` again and again.

Reply:
Oh yes, I'll fix this.